sretoolbox/container/image.py

```diff
@@ -113,11 +120,27 @@ def __init__(self, url, tag_override=None, username=None, password=None,
 
         self._cache_tags = None
         self._cache_manifest = None
+        self._cache_content_type = None
+
+    @property
+    def content_type(self):
+        """
+        Property to return the Content-Type header from the manifest retrieval.
+        It caches the result.
+        """
+        if self._cache_content_type is None:
+            manifest = self._get_manifest()
```

Comment:
We should call `self.manifest` here, as it caches the self._get_manifest(). Most of other calls are using `self.manifest`, except for the one in the `digest()` method, which should also use `self.manifest`, I believe.

Reply:
The problem is that we're caching the manifest json, not the whole manifest response with the headers... My first idea was to cache the whole response and then make `manifest` method to return the `json` response and `content_type` just the corresponding header... But the I saw the `digest` method and implemented a similar one. I still like more the idea of caching the whole response, so I will give it a try.

Reply:
Tried in 9dfa880